### PR TITLE
Add `Coil:*:DX:VariableSpeed` as valid `AirflowNetwork:Distribution:Component:Coil`

### LINKED ISSUE #10671: Add Coil:Cooling:DX:VariableSpeed and Heating to AirflowNetwork:Distribution:Component:Coil
Issue overview
--------------
Coil:Cooling:DX:VariableSpeed and Coil:Heating:DX:VariableSpeed are missing from the valid coil types for AirflowNetwork:Distribution:Component:Coil.
```
AirflowNetwork:Distribution:Component:Coil,
      \min-fields 4
      \memo This object defines the name of a coil used in an air loop.
 A1 , \field Coil Name
      \required-field
      \type object-list
      \reference AirflowNetworkComponentNames
      \object-list AFNCoilNames
      \note Enter the name of a cooling or heating coil in the primary Air loop.
 A2 , \field Coil Object Type
      \required-field
      \type choice
      \key Coil:Cooling:DX
      \key Coil:Cooling:DX:SingleSpeed
      \key Coil:Cooling:DX:TwoSpeed
      \key Coil:Heating:Fuel
      \key Coil:Heating:Electric
      \key Coil:Heating:DX:SingleSpeed
      \key Coil:Cooling:Water
      \key Coil:Heating:Water
      \key Coil:Cooling:Water:DetailedGeometry
      \key Coil:Cooling:DX:TwoStageWithHumidityControlMode
      \key Coil:Cooling:DX:MultiSpeed
      \key Coil:Heating:DX:MultiSpeed
      \key Coil:Heating:Desuperheater
      \key Coil:Heating:Electric:MultiStage
      \key Coil:Heating:Gas:MultiStage
      \note Select the type of coil corresponding to the name entered in the field above.
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus 22.2
 - Helpdesk ticket number 16742.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lymereJ**
Pull request overview
---------------------
 - Fixes #10671 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp [hunk 1]
```diff
@@ -10622,6 +10620,26 @@ namespace AirflowNetwork {
...
+                } else if (SELECT_CASE_var == "COIL:HEATING:DX:VARIABLESPEED") {
+                    ValidateComponent(
+                        m_state, "Coil:Heating:DX:VariableSpeed", DisSysCompCoilData(i).name, IsNotOK, format(RoutineName) + CurrentModuleObject);
+                    ++MultiSpeedHPIndicator;
+                    if (IsNotOK) {
+                        ErrorsFound = true;
+                    } else {
+                        SetDXCoilAirLoopNumber(m_state, DisSysCompCoilData(i).name, DisSysCompCoilData(i).AirLoopNum);
+                    }
+
```
Comment (lymereJ) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:10642]: Add variable speed coils as a possible coil choice.

File: doc/input-output-reference/src/overview/group-airflow-network.tex [hunk 2]
```diff
@@ -2156,8 +2156,12 @@ \subsubsection{Inputs}\label{inputs-2016-06-16}
   \hyperref[coilcoolingdxtwostagewithhumiditycontrolmode]{Coil:Cooling:DX:TwoStageWithHumidityControlMode}
 \item
   \hyperref[coilcoolingdxmultispeed]{Coil:Cooling:DX:MultiSpeed}
+\item
+  \hyperref[coilcoolingdxvariablespeed]{Coil:Cooling:DX:VariableSpeed}
 \item
   \hyperref[coilheatingdxmultispeed]{Coil:Heating:DX:MultiSpeed}
+\item
+  \hyperref[coilheatingdxvariablespeed]{Coil:Heating:DX:VariableSpeed}
```
Comment (lymereJ) [inline doc/input-output-reference/src/overview/group-airflow-network.tex:2164]: Update the documentation.

**Review by jasondegraw — APPROVED** [review]
This looks good to me.

**Review by Myoldmopar — APPROVED** [review]
OK, this makes sense.  Just adding them as valid entries, not having to actually modify any simulation code really.  :+1:

**Comment by Myoldmopar** [comment]
Pulled develop in and tested it, all happy.  Those 2 Linux results are likely just the random issues we've seen recently on Linux.  Merging this.  Thanks @lymereJ and @jasondegraw

**Pull request merged**